### PR TITLE
Update source / destination middleware docs

### PR DESCRIPTION
**Pull request by silesky**
- Document * API for destination middleware
- Add better examples for developers.

### CODE REVIEW

File: src/connections/sources/catalog/libraries/website/javascript/middleware.md [hunk 1]
```diff
@@ -20,42 +20,72 @@ The first function (Source Middleware) allows you to manipulate the payload and
...
+  }
+  next(payload)
+});
 ```
 
 ## Using Destination Middlewares
 
-The function signature for creating Destination Middleware also has three parameters:
+
+To add source middleware, use the following API:
```
Comment (forstisabella) [inline src/connections/sources/catalog/libraries/website/javascript/middleware.md:59]:
```suggestion
To add destination middleware, use the following API:
```

**Review by forstisabella — APPROVED** [review]
lgtm!

**Review by cmastr — APPROVED** [review]

File: src/connections/sources/catalog/libraries/website/javascript/middleware.md (hunk 1 above)
Comment (silesky) [reply src/connections/sources/catalog/libraries/website/javascript/middleware.md:59]: Oops -- yep... wait, did this get merged already?

Comment (forstisabella) [reply src/connections/sources/catalog/libraries/website/javascript/middleware.md:59]: ah oops! I'll open another PR to include this @silesky

**Pull request merged**